application/src/main/resources/log4j2.xml

```diff
@@ -4,18 +4,27 @@
         <Console name="Console" target="SYSTEM_OUT">
             <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
         </Console>
-        <RollingFile name="File" fileName="logs/app.log" filePattern="logs/app-%d{dd-MM-yyyy}.log.gz"
+        <RollingFile name="File" fileName="logs/app.log"
+                     filePattern="logs/app-%d{dd-MM-yyyy}.log.gz"
                      ignoreExceptions="false">
-            <PatternLayout pattern="%d{yyy-MM-dd HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
+            <PatternLayout charset="UTF-8"
+                           pattern="%d{yyy-MM-dd HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
             <Policies>
                 <SizeBasedTriggeringPolicy size="10 MB"/>
             </Policies>
         </RollingFile>
+        <Http name="Http" url="http://${env:TJ_LOG_TARGET:-localhost}:5050/rest/api/logs">
+            <JsonLayout/>
+        </Http>
+        <Async name="AsyncHttp">
```

Comment:
Just wondering how an http appender can be made async unlike disk IO you can flush data in a batch it still has to be over multiple http calls. If it doesnt work as expected the cost of accumulated http calls can slow down the operations of the bot. Also what happens if the http endpoint becomes unavailable?

Reply:
If the Endpoint isn't available, an error get's logger to console but not to file.

@I-Al-Istannen wanted the Async Wrapper around the HTTP Appender

Reply:
https://logging.apache.org/log4j/2.x/manual/appenders.html
It just spins up a new thread as directly logging to HTTP would make log calls very expensive. The Async appender allows us to specify what should happen when the queue fills up - block or discard messages.